forte/pipeline_component.py

```diff
@@ -31,6 +31,14 @@ class PipelineComponent(Generic[PackType]):
     def __init__(self):
         self.resources: Resources = Resources()
         self.configs: Config = Config({}, {})
+        self._check_type_consistency = False
+
+    def enforce_consistency(self):
```

Comment:
You should probably add a boolean input just like the `pipeline`: `    def enforce_consistency(self, enforce: bool = True):`